client/src/components/CheckboxSelector.js

```diff
@@ -0,0 +1,55 @@
+import './CheckboxSelector.scss';
+
+export class CheckboxSelector extends React.Component {
+  constructor(props){
+    super(props);
+  }
+  componentDidUpdate(){
+  }
+  render(){
+    const { items, handleToggle } = this.props;
+    return (
+      _.map(items, (item, key) =>
+        <Checkbox key={key} toggleHandler={handleToggle} {...item}/>
+      )
+    );
+  }
+}
+
+export class Checkbox extends React.Component {
+  constructor(props){
+    super(props);
+    this.state = {
+      checked: props.active,
+    };
+  }
+  render(){
+    const {id, label, color} = this.props;
+    const checked = this.state.checked;
+    const onClickHandler = () => {
+      const newChecked = !this.state.checked;
+      this.props.toggleHandler(this.props.id,newChecked);
+      this.setState({checked: newChecked});
```

Comment:
Since this component has to use a callback to have any real behaviour, there's always going to be some parent component that's keeping track of its own copy of the checkbox items and their checked/unchecked state, right? I'd suggest making this component stateless and just using the `active` prop on every render.

Reply:
:+1: